vertx-grpc-common/src/main/java/io/vertx/grpc/common/impl/GrpcMessageImpl.java

```diff
@@ -38,15 +39,31 @@ public Buffer payload() {
   }
 
   public static Buffer encode(GrpcMessage message) {
-    ByteBuf bbuf = ((BufferInternal)message.payload()).getByteBuf();
+    ByteBuf bbuf = ((BufferInternal) message.payload()).getByteBuf();
     int len = bbuf.readableBytes();
     boolean compressed = !message.encoding().equals("identity");
+    // it is worthy to just copy here, 'cause composite (heap) buffers are slower to be sent to the wire or copied
+    if (len <= 128) {
+      int totalBytes = 5 + len;
+      // let's use vertx buffer here because it doesn't have any atomic release, if unpooled
+      ByteBuf fullMsg = VertxByteBufAllocator.DEFAULT.heapBuffer(totalBytes, totalBytes);
```

Comment:
Maybe we would like to use pooled direct buffers here instead, wdyt @vietj ?
It would save the transport later to copy it into a direct one :/